src/core/annotation.js

```diff
@@ -661,6 +677,63 @@ class WidgetAnnotation extends Annotation {
     return !!(this.data.fieldFlags & flag);
   }
 
+  /**
+   * Parse widget annotation appearance and extract font information.
+   *
+   * @public
+   * @memberof WidgetAnnotation
+   * @param {number} flag - Hexadecimal representation for an annotation
```

Comment:
Ok, cool, sounds great. I just not 100% sure for private vs public because this method returns promise and I call it after widget annotation creation.

Reply:
If you implement the comment above, this can become private because you would not need to call it manually anymore. Instead, it would be called in the constructor of the object itself, therefore the other code does not need to know about it.

Reply:
But how can I return promise right from the constructor? Can you show me some code snippet, maybe I just not very well familiar with promises syntax

Reply:
Now I see the problem here. Indeed, returning a promise from the constructor is not a good pattern. Your way of doing this is better then, but I'm still a bit on the fence about letting other code trigger the fetching of the appearance data; it seems like something the class should handle itself, but I'm not sure if that's possible. For now you can leave the code like you have it already; perhaps if the updated patch is pushed (with should already be smaller given that most of the comments are adressed) we'll get new ideas or we'll just leave it like this.